### PR TITLE
Add script to bootstrap cpp build.

### PR DESCRIPTION
**Pull request by geirst**
@bjorncs please review
@vekterli @toregge @aressem FYI

### CODE REVIEW

File: bootstrap-cpp.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    usage
+    exit 1
+fi
+
+mkdir -p "${BUILD_DIR}"
+
+source /opt/rh/devtoolset-6/enable || true
+cd "${SOURCE_DIR}"
+sh ./bootstrap.sh full
+cd "${BUILD_DIR}"
```
Comment (bjorncs) [inline bootstrap-cpp.sh]: This statement will most likely fail if both `SOURCE_DIR` and `BUILD_DIR` are relative paths. Replace both instances of _cd_ with _pushd_/_popd_.

**Review by bjorncs — CHANGES_REQUESTED** [review]

**Comment by geirst** [comment]
@toregge or @bjorncs please review again

File: bootstrap-cpp.sh [hunk 2]
```diff
@@ -29,6 +29,5 @@ cmake3 \
     -DEXTRA_LINK_DIRECTORY="/opt/vespa-boost/lib;/opt/vespa-libtorrent/lib;/opt/vespa-zookeeper-c-client/lib;/opt/vespa-cppunit/lib;/usr/lib64/llvm3.9/lib" \
     -DEXTRA_INCLUDE_DIRECTORY="/opt/vespa-boost/include;/opt/vespa-libtorrent/include;/opt/vespa-zookeeper-c-client/include;/opt/vespa-cppunit/include;/usr/include/llvm3.9" \
     -DCMAKE_INSTALL_RPATH="/opt/vespa/lib64;/opt/vespa-boost/lib;/opt/vespa-libtorrent/lib;/opt/vespa-zookeeper-c-client/lib;/opt/vespa-cppunit/lib;/usr/lib/jvm/java-1.8.0/jre/lib/amd64/server;/usr/include/llvm3.9" \
-    -DCMAKE_BUILD_RPATH=/opt/vespa/lib64 \
     -DVALGRIND_UNIT_TESTS=no \
```
Comment (toregge) [inline bootstrap-cpp.sh]: This line (setting VALGRIND_UNIT_TESTS to no)  isn't needed either, default is FALSE, cf. build_settings.cmake.

**Review by bjorncs — APPROVED** [review]

File: bootstrap-cpp.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    exit 0
+else
+    echo "Wrong number of arguments: expected 2, was $#" >&2
+    usage
+    exit 1
+fi
+
+mkdir -p "${BUILD_DIR}"
+
+source /opt/rh/devtoolset-6/enable || true
```
Comment (toregge) [inline bootstrap-cpp.sh]: The caller should have done the source, since the changes are likely needed in a wider scope.

Comment (geirst) [inline bootstrap-cpp.sh]: /opt/rh/devtoolset-6/root/usr/lib64 is prepended to LD_LIBRARY_PATH, but we do not use any of the libs there. python path is also set, but we do not use this either. Conclusion: We can keep it as is for now.

**Pull request merged**